### PR TITLE
[#15] Update to Hono 2.1

### LINKED ISSUE #15: Update to Hono 2.1
This requires https://github.com/eclipse/hono/issues/3369 to be resolved.

### PR DESCRIPTION
**Pull request by calohmn**
Resolves #15.

### CODE REVIEW

File: protocol-gateway/mqtt-protocol-gateway-template/pom.xml [hunk 1]
```diff
@@ -66,7 +66,31 @@
     <dependencies>
         <dependency>
```
[the remark this answers is not in the document]
Comment (calohmn) [reply protocol-gateway/mqtt-protocol-gateway-template/pom.xml:67]: @sonatype-lift ignore

File: protocol-gateway/mqtt-protocol-gateway-template/pom.xml [hunk 2]
```diff
@@ -66,7 +66,31 @@
...
             <groupId>org.eclipse.hono</groupId>
-            <artifactId>hono-client</artifactId>
+            <artifactId>hono-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.hono</groupId>
+            <artifactId>hono-client-common</artifactId>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (calohmn) [reply protocol-gateway/mqtt-protocol-gateway-template/pom.xml:75]: @sonatype-lift ignore

File: protocol-gateway/mqtt-protocol-gateway-template/pom.xml [hunk 3]
```diff
@@ -66,7 +66,31 @@
     <dependencies>
         <dependency>
             <groupId>org.eclipse.hono</groupId>
-            <artifactId>hono-client</artifactId>
+            <artifactId>hono-core</artifactId>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (calohmn) [reply protocol-gateway/mqtt-protocol-gateway-template/pom.xml:71]: @sonatype-lift ignore

File: protocol-gateway/mqtt-protocol-gateway-template/pom.xml [hunk 4]
```diff
@@ -66,7 +66,31 @@
...
+            <groupId>org.eclipse.hono</groupId>
+            <artifactId>hono-client-common</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.hono</groupId>
+            <artifactId>hono-client-amqp-connection</artifactId>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (calohmn) [reply protocol-gateway/mqtt-protocol-gateway-template/pom.xml:79]: @sonatype-lift ignore

File: protocol-gateway/mqtt-protocol-gateway-template/pom.xml [hunk 5]
```diff
@@ -66,7 +66,31 @@
...
+            <groupId>org.eclipse.hono</groupId>
+            <artifactId>hono-client-application</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.hono</groupId>
+            <artifactId>hono-client-application-amqp</artifactId>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (calohmn) [reply protocol-gateway/mqtt-protocol-gateway-template/pom.xml:91]: @sonatype-lift ignore

File: protocol-gateway/mqtt-protocol-gateway-template/pom.xml [hunk 6]
```diff
@@ -66,7 +66,31 @@
...
+            <groupId>org.eclipse.hono</groupId>
+            <artifactId>hono-client-amqp-common</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.hono</groupId>
+            <artifactId>hono-client-application</artifactId>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (calohmn) [reply protocol-gateway/mqtt-protocol-gateway-template/pom.xml:87]: @sonatype-lift ignore

**Comment by calohmn** [comment]
I've set the Lift findings concerning the `netty-handler@4.1.78.Final` dependency in the Hono components to be ignored here.

**Comment by calohmn** [comment]
(Sonatype Lift is not analyzing the Java sources here. I'll add another PR to enable this along with some fixes for Lift findings.)

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**